### PR TITLE
ci: unique remote github repo

### PR DESCRIPTION
**Pull request by oldgiova**
@alfrunes I saw that in the same runner, sometime old github remote could "survive", and could be from past try and tests, so there is no guarantees the `github` remote is the good one.
This PR address this behavior by always setting a new unique github remote `github-${CI_JOB_ID}`, and removes it in the `after_script`. Successfully tested in a [sample pipeline](https://gitlab.com/Northern.tech/northerntechhq/nt-boilerplate-pipeline/-/jobs/8070161234)

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -856,25 +856,27 @@ changelog:
...
-        && git reset --hard FETCH_HEAD
-        && wget --output-document cliff.toml https://raw.githubusercontent.com/mendersoftware/mendertesting/master/utils/cliff.toml
+    - git fetch github-${CI_JOB_ID} pull/${RELEASE_PLEASE_PR}/head
+    - git reset --hard FETCH_HEAD
+    - wget --output-document cliff.toml https://raw.githubusercontent.com/mendersoftware/mendertesting/master/utils/cliff.toml
         && git cliff --bump --output mender/CHANGELOG.md --github-repo ${GITHUB_REPO_URL}
         && git add mender/CHANGELOG.md
         && git commit --amend -s --no-edit
-        && git push github --force
+        && git push github-${CI_JOB_ID} --force
```
Comment (alfrunes) [inline .gitlab-ci.yml:871]: Could we replace all `&&` with `-`?
Logically it's the same, only the latter is much easier to debug. Instead of seeing the first command and "collapsed multiline command" on failure, you will see the actual command that failed.

**Review by alfrunes — COMMENTED** [review]
I still don't quite get it; it's the same remote. Even if the local git repository is cached, `git fetch` still fetches the upstream PR.

File: .gitlab-ci.yml (hunk 1 above)
Comment (oldgiova) [reply .gitlab-ci.yml:871]: The reason for all this `&&` is because you have a size limit of the array script; if I good remember is 10, but let me try again.

**Comment by oldgiova** [comment]
> I still don't quite get it; it's the same remote. Even if the local git repository is cached, `git fetch` still fetches the upstream PR.

From my tests, I found an old remote from a former test with the same name, but incorrectly formatted. So we have no 100% guarantees that the remote is actually the same/
